lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,255 @@
+from __future__ import absolute_import
+
+import re
+import subprocess
+from subprocess import Popen, PIPE
+import datetime
+import random
+from collections import namedtuple
+import traceback
+
+import gdb
+
+
+def parse_raw_symlist(syml):
+    symre = r' ([^ ]*)$'
+    symc = re.compile(symre)
+    for line in syml.split('\n'):
+        m = symc.search(line)
+        if m:
+            yield(m.group(1))
+
+
+def extract_symbol_list_from_so(file_name):
+    try:
+        symlist = subprocess.check_output(['nm', '-C', file_name])
+    except subprocess.CalledProcessError:
+        print('nm -C {} failed'.format(file_name))
+        raise
+    except OSError:
+        print('no usable "nm" found, cannot continue')
+        raise
+    assert symlist, 'no symbols in file {}'.format(file_name)
+    return symlist
+
+
+def extract_symbols_from_raw_list(rs):
+    '''
+    Saves the entire symbol names as a single long string
+    so we can say something like:
+        if symbol in symbol_list
+    '''
+    parsed_list = []
+    for sym in rs:
+        sym_name = subprocess.check_output(['c++filt', sym])
+        assert sym_name, 'c++filt didn\'t demangle {}'.format(sym)
+        parsed_list.append(sym_name)
+    return ''.join(parsed_list)
```

Comment:
Why not returning a list? You can still do if foo in bar with a list.

Reply:
but then I'll have to match the element exactly while with strings I can have partial match (that is what I want in this case): The user doesn't have to know exactly the symbol name

Reply:
Ah, good point. But you could also find alse positives so maybe it should be `' '.join(...)` with a space?
Also you should document why you do that.

Reply:
yup, well spotted